index.js

```diff
@@ -201,9 +247,17 @@ var znFilterMatcher = (function() {
 			// Rule contains "and"/"or" key - nested filter
 			return recordMatchesFilter(record, rule);
 		}
-
 		if (rule.filter !== undefined) {
-			throw new Error("Subfilter matching is not supported");
+			var subRecord = record[rule.attribute];
+			if (subRecord && (noKeys(subRecord) || anyKeysLike(subRecord, /^field/))) {
+				if (!recordMatchesFilter(subRecord, rule.filter)) {
+					return false;
+				}
+				return true;
+			} else {
+				throw new Error("Subfilter matching is not supported");
```

Comment:
@dwmcnelis not sure I understand these conditions for when it should recurse vs throw an error. 

1.) You could hypothetically have a record with just standard attributes like folder id and record title, and a subfilter that only matches on these attributes. So it would fail this condition `anyKeysLike(subRecord, /^field/)` and cause `recordMatchesFilter` to not be called. It seems inconsistent to impose this requirement on a subrecord when its not required at the top level. 
2.) If the primary record doesn't have a linked record set, i.e. if the value returned from the Zengine API is null, then it seems like instead of throwing an error, it should return false.

Separately, I think it would be really helpful to add unit tests for this. We can pretty easily port these over from anglerfish - https://github.com/Wizehive/anglerfish/blob/stage/tests/unit/services/zn-filter-matcher-spec.js - and then add some for subfiltering. These tests can also double as documentation for developers trying to figure out how they need to format their records in order to make subfiltering work.